homeassistant/components/reolink/host.py

```diff
@@ -151,4 +167,135 @@ async def disconnect(self):
 
     async def stop(self, event=None):
         """Disconnect the API."""
+        await self.unregister_webhook()
         await self.disconnect()
+
+    async def subscribe(self) -> None:
+        """Subscribe to motion events and register the webhook as a callback."""
+        if self.webhook_id is None:
+            await self.register_webhook()
+
+        if self._api.subscribed:
+            _LOGGER.debug(
+                "Host %s: is already subscribed to webhook %s",
+                self._api.host,
+                self._webhook_url,
+            )
+            return
+
+        if await self._api.subscribe(self._webhook_url):
+            _LOGGER.info(
+                "Host %s: subscribed successfully to webhook %s",
+                self._api.host,
+                self._webhook_url,
+            )
+        else:
+            raise ReolinkWebhookException(
+                f"Host {self._api.host}: webhook subscription failed"
+            )
+
+    async def renew(self) -> None:
+        """Renew the subscription of the motion events (lease time is set to 15 minutes)."""
+
+        if not self._api.subscribed:
+            _LOGGER.debug(
+                "Host %s: requested to renew a non-existing Reolink subscription, trying to subscribe from scratch",
```

Comment:
Please breaking long strings and comments around 88 characters per line.

Reply:
done